src/Swarm/TUI/View.hs

```diff
@@ -621,7 +621,9 @@ drawModal s = \case
         ]
   DescriptionModal e -> descriptionWidget s e
   QuitModal -> padBottom (Pad 1) $ hCenter $ txt (quitMsg (s ^. uiState . uiMenu))
-  GoalModal -> GR.renderGoalsDisplay (s ^. uiState . uiGoal)
+  GoalModal ->
+    GR.renderGoalsDisplay (s ^. uiState . uiGoal) $
+      view scenarioDescription . fst <$> s ^. uiState . scenarioRef
```

Comment:
Untested, but would something like this work?

```suggestion
      s ^? uiState . scenarioRef . _Just . _1 . scenarioDescription
```

Reply:
Eh, I'm not really a fan of the `_Just` and `_1` lenses when they can be avoided.  I recognize they are not technically "partial functions" here, but the explicit indexing and `Just` are evocative of partiality in a way I find unpalatable :stuck_out_tongue:

Reply:
OK, that's fine!